scheduler/filter/port.go

```diff
@@ -20,23 +20,51 @@ func (p *PortFilter) Name() string {
 
 // Filter is exported
 func (p *PortFilter) Filter(config *dockerclient.ContainerConfig, nodes []cluster.Node) ([]cluster.Node, error) {
-	for _, port := range config.HostConfig.PortBindings {
-		for _, binding := range port {
+	if config.HostConfig.NetworkMode == "host" {
```

Comment:
Minor nit: This function is a bit complex, could you break it down to two functions, `filterHost` and `filterBridge`, and in `Filter` do something like:

```
if config.HostConfig.NetworkMode == "host" {
  return filterHost(...)
} else {
  return filterBridge(...)
}
```